components/script/dom/bluetoothremotegattserver.rs

```diff
@@ -134,6 +149,14 @@ impl AsyncBluetoothListener for BluetoothRemoteGATTServer {
         match response {
             // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattserver-connect
             BluetoothResponse::GATTServerConnect(connected) => {
+                // Step 5.2.3
+                if self.Device().is_represented_device_null() {
+                    if let Err(e) = self.garbage_collect_the_connection() {
+                        promise.reject_error(promise_cx, Error::from(e));
+                    }
+                    promise.reject_error(promise_cx, Error::Network);
```

Comment:
The spec says to abort the steps here, otherwise we'll also resolve the promise.